packages/cursorless-engine/src/processTargets/targets/DestinationImpl.ts

```diff
@@ -197,8 +197,12 @@ function getIndentationString(editor: TextEditor, range: Range) {
       continue;
     }
 
-    if (line.firstNonWhitespaceCharacterIndex < length) {
-      length = line.firstNonWhitespaceCharacterIndex;
+    const trimmedPosition = line.isEmptyOrWhitespace
+      ? line.rangeTrimmed.end
+      : line.rangeTrimmed.start;
+
+    if (trimmedPosition.character < length) {
```

Comment:
why has this gotten more awkward? I feel that maybe the semantics of `trimmedRange` are maybe the opposite of what they should be? Looks like when we had the two separate attributes, they kinda formed a backward range when it was empty

Reply:
update from meet-up: it appears that the new api has exposed existing awkwardness here, and with some thought, we might be able to simplify / clarify this whole loop body